hera_cal/io.py

```diff
@@ -579,9 +579,19 @@ def _determine_blt_slicing(self):
     def _determine_pol_indexing(self):
         '''Determine the mapping between polnums and indices
         in the polarization axis of the data_array.'''
-        self._polnum_indices = {}
-        for i, polnum in enumerate(self.polarization_array):
-            self._polnum_indices[polnum] = i
+        self._polnum_indices = {
+            polnum: i for i, polnum in enumerate(self.polarization_array)
+        }
+        pols = [polnum2str(polnum, x_orientation=self.x_orientation) for polnum in self.polarization_array]
+        self._polstr_indices = {}
+        # Add upper-case indices as well, so we don't need to use .lower() on input
+        # keys (for which there can be many tens of thousands).
+        for pol in pols:
+            indx = self._polnum_indices[polstr2num(pol, x_orientation=self.x_orientation)]
+            self._polstr_indices[pol.lower()] = indx
+            self._polstr_indices[pol.upper()] = indx
+            self._polstr_indices[pol[0].upper() + pol[1].lower()] = indx
```

Comment:
If for some weird reason the pol is a single char, this would break. Not sure if it broke before, so perhaps worth checking?

Instead of handling edge cases like this, what if we used dictionary default behavior? For example:

`self._polstr_indices[key[2], self._polnum_indices[polstr2num(key[2], x_orientation=self.x_orientation)]`

Reply:
Yeah, good idea.

Reply:
I left the overall lower/upper, but not the weird cases where one one letter is upper and the other lower.